packages/react-client/src/ReactFlightClientStream.js

```diff
@@ -46,6 +47,10 @@ function processFullRow(response: Response, row: string): void {
       resolveModule(response, id, row.slice(colon + 2));
       return;
     }
+    case 'D': {
```

Comment:
Here's what I had in mind for typed arrays:

```
- C - Int8Array (char)
- c - Uint8Array (unsigned char)
- Z - Uint8ClampedArray (unsigned clamped char)
- S - Int16Array (short)
- s - Uint16Array (unsigned short)
- L - Int32Array (long)
- l - Uint32Array (unsigned long)
- F - Float32Array (float)
- D - Float64Array (double)
- N - BigInt64Array (number)
- n - BigUint64Array (unsigned number)
- A - ArrayBuffer
```

As you can see `D` represents `double` in this pattern. I'm not 100% sure this plan will work but can we pick another character?

Reply:
It's possible we just end up encoding this in the reference instead though.

Reply:
I think `!` is a nice alternative

Reply:
Punctuations could apply to everything. Why is this one special? It can't be anything that's in the first character of JSON since that's implicitly a model. So that eliminates `[`, `{`, `"` and numbers. So we quickly run out of punctuators. And since punctuators generally means JSON it's extra confusing.

Let's just pick a name. Although my other question is if we should just encode preload/preconnect/preinit etc into this character instead of the payload.

Reply:
I picked `!` for emphasis that these things should be acted on immediately. A letter is fine though :)

Reply:
We decided to go with `H` + char to represent Hints of various types.

`HD` -> prefetchDNS
`HC` -> preconnect
`HL` -> preload
`HI` -> preinit

can support more (up to alphabet limit) which should be more than enough.
Also note there is no longer a general "Directive" concept. these are hints. This will be helpful in allowing for deduping, and specialized re-ordering of hints